src/poliastro/core/spheroid_location.py

```diff
@@ -177,11 +177,13 @@ def cartesian_to_ellipsoidal(_a, _c, x, y, z):
     e2_ = e2 / (1 - e2)
     p = np.sqrt(x ** 2 + y ** 2)
     th = np.arctan(z * _a / (p * _c))
-    lon = np.arctan(y / x)
+    lon = np.arctan2(y, x)  # Use `arctan2` so that lon lies in the range: [-pi, +pi]
     lat = np.arctan((z + e2_ * _c * np.sin(th) ** 3) / (p - e2 * _a * np.cos(th) ** 3))
 
     v = _a / np.sqrt(1 - e2 * np.sin(lat) ** 2)
-    h = p / np.cos(lat) - v
-    h = z / np.sin(lat) - (1 - e2) * v
+    if lat == np.pi / 2 or lat:
```

Comment:
I know that this evaluates to `False` if `lat == 0`, but could we make it explicit? Many Python programmers don't know this...